src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -1571,12 +1571,17 @@ void RequireDynamicallyAccessedMembers (ref ReflectionPatternContext reflectionC
 				} else if (uniqueValue is SystemTypeValue systemTypeValue) {
 					MarkTypeForDynamicallyAccessedMembers (ref reflectionContext, systemTypeValue.TypeRepresented, requiredMemberTypes);
 				} else if (uniqueValue is KnownStringValue knownStringValue) {
-					TypeDefinition foundType = _context.TypeNameResolver.ResolveTypeName (knownStringValue.Contents)?.Resolve ();
+					TypeReference typeRef = _context.TypeNameResolver.ResolveTypeName (knownStringValue.Contents);
+					TypeDefinition foundType = typeRef?.Resolve ();
```

Comment:
This will be a problem when arrays get here - since for some weird Cecil reason calling `Resolve` on `T[]` returns resolved `T` (it drops the "array" part). As per the discussion in the array issue, we definitely need to mark the array itself at least.

Reply:
We're now passing the typref to `MarkType`, this shouldn't be a problem anymore.